### PR TITLE
Set up association between classes

### PR DESCRIPTION
**Pull request by rica213**
In this milestone, here are the things I have done: 

- Create a class Classroom
- Create the has-many/belongs-to relationship between Classroom and Student : a classroom has many students and a student belongs to a classroom
- Create a class Book 
- Create a class Rental
- Create the many-to-many relationship between Person and Book using the intermediate class Rental: a book has many rentals, a rental belongs to a book, a person has many rentals, and a rental belongs to a person.
- Modify the constructor of Rental so Book and Person are set in it.
- Update readme file

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require './student'
```
Comment (rloterh) [inline classroom.rb:1]: - [ ] [OPTIONAL]
Let's be reminded that by convention, it's a good practice to use `require_relative` classes in the same source and use `require` for external modules or packages. Kindly ensure to improve upon this aspect.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './rental'
+
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (rloterh) [inline book.rb:10]: - [ ] A _rental_ can have many _books_, and _books_ can be subjected to many _rentals_. In order for us to establish a _many-to-many_ association between _rentals_ and _books_ (let's bear in mind that the association between _person_ and _rentals_ is also _many-to-many_), let's define a method for adding rentals within the book object similar to the example below:

```ruby
class Book
  attr_accessor :title, :author, :rentals
  .
  .

  def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book
```
Comment (rloterh) [inline rental.rb:2]: - [ ] Let's mark `book` and `person` with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ to it. Kindly ensure to improve upon this aspect.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Larissa, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁  🌴 🚦 🚗 🏫  🏡   _🏃 

## Highlights 👍  
- Good job so far establishing associations for a book library objects! 
- ReadMe looks good!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- Generally, it's good idea that the subheading text _Author1_ is replaced with your full name in your ReadMe.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  def rentals=(rental)
+    @rentals.append(rental)
+  end
```
Comment (BregornOriginal) [inline book.rb:15]: Good work creating a method to add a rental, but there are some typos in it.

- As was requested by the previous reviewer, kindly follow the example that he give you.
- There's no need to put an `=` when you define a method.
- The parameters that you will receive on the method `rentals` will be `(person, date)`, and when you create a Rental, you need to pass this parameter.

Here's a [LINK ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)to the relations for many-to-many as refresh

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
### Hi @rica213  

### Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


giphy (3)

 ## Highlights

✅ Good commit history 💯 
✅ Good Pull Request description

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frank1738 — APPROVED** [review]
Hi @rica213  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Organized code

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**